students/becky_larson/lesson09/donor_models.py

```diff
@@ -0,0 +1,293 @@
+        donations = ",".join([f"{d:.02f}" for d in self._donations])
+        return f"{self._donor_name},{donations}"
+
+    def __lt__(self, other):
+        """
+        Return if this donor is less than another donor based on donations.
+        """
+        tuple_self = (self.total_donations, self.avg_donation)
+        tuple_other = (other.total_donations, other.avg_donation)
+        return tuple_self < tuple_other
+
+    @classmethod
+    def from_string(cls, input):
+        """
+        Instantiate a donor from a string.
+        :input:  The string to parse into a donor.
+        """
+        input = input.split(',')
+        name = input[0]
+        donations = [float(donation) for donation in input[1:]]
+        self = cls(name, donations)
+        return self
+
+    def __repr__(self):
+        """
+        :return: string of self instantiation
+        """
+        return f"Donor({self._donor_name!r}, {self._donations!r})"
+
+    def sort_key(self):
+        return self.last_name, self.first_name, self.donations
+
+    def sort_by_donations(self):
+        return self.total_donations
+
+
+class DonorCollection:
+    """DonorCollection: holds all of the donor objects
+    Methods to add a new donor, search for a given donor, etc.
+    """
+    def __init__(self, donors=()):
+        """
+        DonorCollection
+        :param donors: dict of Donor object(s)
+        """
+
+        self._donors = dict()
+
+        if hasattr(donors, '__iter__') and not isinstance(donors, str):
+            for donor in donors:
+                if isinstance(donor, Donor):
+                    self._donors[donor.donor_name] = donor
+                else:
+                    raise TypeError("object is not of type Donor")
+        self.append(donors)
+
+    def __len__(self):
+        """
+        Return the length of the donor list.
+        """
+        return len(self._donors)
+
+    @property
+    def donors(self):
+        """
+        getter for tuple of Donor objects
+        :return: donors
+        """
+        return tuple(self._donors)
+
+    def __getitem__(self, donor_name: str):
+        """
+        return Donor object using [] operator
+        :param donor_name: key as name of donor
+        :return: Donor object donations
+        """
+        return self._donors[donor_name]
+
+    def list_donors(self):
+        returned_list = []
+        for donor in self._donors:
+            returned_list.append(donor)
+        return returned_list
+
+    def is_donor_new(self, donor_name):
+        is_donor_new = True
+        for donor in self._donors:
+            if donor == donor_name:
+                is_donor_new = False
+                break
+
+        return is_donor_new
+
+    def append(self, donor: Donor):
+        """
+        add Donor object to collection
+        :param donor: Donor object or sequence of Donor objects
+        :return: None
+        """
+        if hasattr(donor, '__iter__'):
+            for _ in donor:
```

Comment:
`_` is usually used to indicate an unused variable, since you're actually using it you should use a real variable name.